src/common/components/hosts/EditHostForm.tsx

```diff
@@ -104,14 +108,26 @@ const EditHostForm: React.FC<EditHostFormProps> = ({
                   {status.error.message}
                 </Alert>
               )}
-              <StaticTextField name="discoveredHostname" label="Discovered Hostname">
+              <Alert
+                variant={AlertVariant.info}
+                title="This name will replace the original discovered hostname."
+                isInline
+              />
+              <StaticTextField name="discoveredHostname" label="Discovered hostname">
                 {hostname || ''}
               </StaticTextField>
               <InputField
-                label="Requested Hostname"
+                label="New hostname"
                 name="hostname"
                 ref={hostnameInputRef}
-                helperText="This name will replace the original discovered hostname after installation."
+                helperText={
+                  <List className="edit-hostname__help-text">
+                    <ListItem>Must be at least 5 characters</ListItem>
+                    <ListItem>Use lower-case letters, digits from 0 to 9 or hyphen (-)</ListItem>
+                    <ListItem>Cannot start or end with a hyphen (-)</ListItem>
+                    <ListItem>Cannot be the word "localhost"</ListItem>
+                  </List>
+                }
```

Comment:
@ShiranHi Patternfly already provides dynamic helper texts.
Please see this example: https://www.patternfly.org/v4/components/helper-text#dynamic-list